pkg/controller/assign/assign_controller.go

```diff
@@ -81,18 +83,25 @@ func (a *Adder) InjectGetPod(getPod func() (*corev1.Pod, error)) {
 	a.GetPod = getPod
 }
 
-func (a *Adder) InjectMutationCache(mutationCache *mutation.System) {
-	a.MutationCache = mutationCache
+func (a *Adder) InjectMutationSystem(mutationCache *mutation.System) {
+	a.MutationSystem = mutationCache
 }
 
 // newReconciler returns a new reconcile.Reconciler.
 func newReconciler(mgr manager.Manager, mutationCache *mutation.System, tracker *readiness.Tracker, getPod func() (*corev1.Pod, error)) *Reconciler {
+	rep, err := mutationReporter.NewStatsReporter()
```

Comment:
It seems like we should be passing the same object to all controllers, rather than creating a new reporter per-controller.